lib/zendesk_apps_support/validations/requests.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'ipaddress'
+
+module ZendeskAppsSupport
+  module Validations
+    module Requests
+      class << self
+        HTTP_REQUEST_METHODS = %w[GET HEAD POST PUT DELETE CONNECT OPTIONS TRACE].freeze
+
+        REQUEST_CALLS = [
+          /\w+\.request\(['"](.*)['"]/i, # ZAF request
+          /\$\.(?:get|post|getJSON)\(['"](.*)['"]/i, # jQuery request
+          /\w+\.open\(['"](?:#{Regexp.union(HTTP_REQUEST_METHODS).source})['"],\s?['"](.*)['"]/i, # XMLHttpRequest
+          /fetch\(['"](.*)['"]/i # fetch
```

Comment:
Should these regex's be case insensitive? `/i`?

Reply:
Yep, intentional (i.e. `$.GET()` is also valid)